### PR TITLE
Show a bar with the events to which the user contributes

### LINKED ISSUE #98: Show inside the translation process that there are active events
Show that this translation or review will contribute to these events. Possible ideas:

Screenshot 2024-02-19 at 12-58-48 Übersetzungen German Development (trunk) GlotPress

### PR DESCRIPTION
**Pull request by amieiro**
This PR adds a bar with the events to which the user contributes.

If the user contributes to up to 3 events, it displays all of them. If she contributes to more, it shows 2 and a text with the number of the other events to which he/she contributes.

### User who does not attend any event

Translations   Galician   Akismet Anti-Spam   GlotPress 2024-02-20 11-21-07

### User attending 1 event

Translations   Galician   Akismet Anti-Spam   GlotPress 2024-02-20 11-17-24

### User attending 2 events

Translations   Galician   Akismet Anti-Spam   GlotPress 2024-02-20 11-16-46

### User attending 3 events

Translations   Galician   Akismet Anti-Spam   GlotPress 2024-02-20 11-19-05

### User attending more than 3 events

Translations   Galician   Akismet Anti-Spam   GlotPress 2024-02-20 11-20-02
 
This PR needs [this PR](https://github.com/GlotPress/GlotPress/pull/1792) to be approved.

Fixes https://github.com/WordPress/wporg-gp-translation-events/issues/98.

### CODE REVIEW

File: wporg-gp-translation-events.php [hunk 1]
```diff
@@ -372,3 +373,80 @@ function () {
...
 	}
 );
+
+/**
+ * Add the active events for the current user before the translation table.
+ *
+ * @return void
+ */
+function add_active_events_current_user(): void {
+	$user_attending_events      = get_user_meta( get_current_user_id(), Route::USER_META_KEY_ATTENDING, true ) ?: array();
```
Comment (psrpinto) [inline wporg-gp-translation-events.php:415]: I think maybe we should consider refactoring [the code that retrieves active events the user is attending](https://github.com/WordPress/wporg-gp-translation-events/blob/trunk/includes/translation-listener.php#L69-L70) to something reusable, as I noticed we need that in several places. 

If we would do that, we would automatically get caching for the query that retrieves active events, as that can be an expensive query.

**Review by psrpinto — COMMENTED** [review]
Looks good. I'll test this once the GlotPress PR gets merged.

File: wporg-gp-translation-events.php [hunk 2]
```diff
@@ -372,3 +373,80 @@ function () {
...
+		}
+
+		$remaining_events = $number_of_events - 2;
+		/* translators: %d: Number of remaining events */
+		$content .= '<span class="remaining-events">' . sprintf( esc_html__( ' and %d more events.', 'gp-translation-events' ), $remaining_events ) . '</span>';
+
+	} else {
+		while ( $user_attending_events_query->have_posts() ) {
+			$user_attending_events_query->the_post();
+			$content .= '<span class="active-events-before-translation-table">' . get_the_title() . '</span>';
```
Comment (akirk) [inline wporg-gp-translation-events.php]: Can we make it a link? Maybe refactor this to use a function.

File: wporg-gp-translation-events.php (hunk 1 above)
Comment (akirk) [reply wporg-gp-translation-events.php:415]: I agree that we should do this! Tracking this in #117

File: wporg-gp-translation-events.php [hunk 3]
```diff
@@ -372,3 +373,80 @@ function () {
...
+
+	$content = '<div id="active-events-before-translation-table" class="active-events-before-translation-table">';
+	/* translators: %d: Number of events */
+	$content .= sprintf( _n( 'Contributing to %d event:', 'Contributing to %d events:', $number_of_events, 'gp-translation-events' ), $number_of_events );
+	$content .= '&nbsp;&nbsp;';
+	if ( $number_of_events > 3 ) {
+		$counter = 0;
+		while ( $user_attending_events_query->have_posts() && $counter < 2 ) {
+			$user_attending_events_query->the_post();
+			$content .= '<span class="active-events-before-translation-table">' . get_the_title() . '</span>';
```
Comment (akirk) [inline wporg-gp-translation-events.php]: Can we make it a link?

**Review by akirk — COMMENTED** [review]
Nice, my only remark is that it looks a bit too much like a warning for my taste but I am curious for other opinions.

**Comment by amieiro** [comment]
> Nice, my only remark is that it looks a bit too much like a warning for my taste but I am curious for other opinions.

What about changing from `--gp-color-attention-emphasis` to `--gp-color-status-waiting-subtle`:

### Current color

image

### New color

image

File: wporg-gp-translation-events.php (hunk 2 above)
Comment (amieiro) [reply wporg-gp-translation-events.php]: Done [here](https://github.com/WordPress/wporg-gp-translation-events/pull/114/commits/091bc7e8a7cdacf6ed278a4a8e727dc479f64f8f).

File: wporg-gp-translation-events.php (hunk 3 above)
Comment (amieiro) [reply wporg-gp-translation-events.php]: Done [here](https://github.com/WordPress/wporg-gp-translation-events/pull/114/commits/091bc7e8a7cdacf6ed278a4a8e727dc479f64f8f).

File: wporg-gp-translation-events.php (hunk 1 above)
Comment (amieiro) [reply wporg-gp-translation-events.php:415]: Ok. I think it is better to address this problem in another PR, as we have an issue to it.

**Comment by amieiro** [comment]
I have added the links to each event in the bubbles and a link to the main page in the "and 2 more events", because the user can see the "Event she is attending" in this page, and we don't have a page with only this information.

I have changed the background color to a lighter one. I prefer to use a GlotPress color, to avoid color inconsistencies if they are changed in the future.

@psrpinto the hook in GlotPress has been merged, so you can pull the `develop` branch from GlotPress and test it.

image

**Review by akirk — APPROVED** [review]
Works well after pulling the latest develop branch of GlotPress. Maybe we can ask a designer later on to take another look at the styling, also in general of the plugin.

**Review by psrpinto — APPROVED** [review]
Tested and it works 👌

**Pull request merged**